OakCityV2/TableViewController.m

```diff
@@ -0,0 +1,216 @@
+//
+//  TableViewController.m
+//  OakCityV2
+//
+//  Created by Josh Green on 6/8/15.
+//  Copyright (c) 2015 Josh Green. All rights reserved.
+//
+
+#import "TableViewController.h"
+#import "AFNetworking/AFNetworking.h"
+#import "DetailViewController.h"
+#import "DetailCell.h"
+#import "Job.h"
+#import "AFNetworking/UIImageView+AFNetworking.h"
+#define LANGUAGE @"PHP"
+#define TOWN @"San+Francisco"
+
+@interface TableViewController ()
+
+@end
+
+@implementation TableViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    //Fetch JSON
+    NSString *urlAsString = [NSString stringWithFormat:@"https://jobs.github.com/positions.json?description=%@&location=%@", LANGUAGE, TOWN];
+    NSURL *url = [NSURL URLWithString:urlAsString];
+    NSURLRequest *request = [NSURLRequest requestWithURL: url];
+    AFHTTPRequestOperation *operation = [[AFHTTPRequestOperation alloc] initWithRequest:request];
+    operation.responseSerializer = [AFJSONResponseSerializer serializer];
+    
+    //Parse JSON
+    [operation setCompletionBlockWithSuccess:^(AFHTTPRequestOperation *operation, id responseObject) {
+        self.jobs  = [NSMutableArray new];
+        NSArray *responseArray = (NSArray *)responseObject;
+        
+        for (NSDictionary *responseDict in responseArray) {
+        Job* job = [[Job alloc] initWithDictionary:responseDict];
+        [self.jobs addObject:job];
+        }
+        
+        if([self.jobs count] > 0) {
+            [self.tableView reloadData];
+        }
+
+        //Alert on failure to fetch JSON
+        else {
+            UIAlertView* alert = [[UIAlertView alloc]
+                                  initWithTitle: @"Failed to retrieve data" message: nil delegate: self
+                                  cancelButtonTitle: @"cancel" otherButtonTitles: @"Retry", nil];
+            [alert show];
+        }
+    }
+     
+     //Alert on fundemental errors
+                                     failure:^(AFHTTPRequestOperation *operation, NSError *error) {
+                                         UIAlertView *aV = [[UIAlertView alloc]
+                                                            initWithTitle:@"Error" message:[error localizedDescription] delegate: nil
+                                                            cancelButtonTitle:@"Ok" otherButtonTitles:nil];
+                                         [aV show];
+                                     }];
+    [operation start];
+    
+}
+
+//Alert View
+- (void)alertView:(UIAlertView *)alertView clickedButtonAtIndex:(NSInteger)buttonIndex {
+    NSString *button = [alertView buttonTitleAtIndex:buttonIndex];
+    
+    if([button isEqualToString:@"Retry"]) {
+        //Fetch JSON
```

Comment:
Application logic based on strings from the user interface is a Very Bad Thing.  This should set off alarms in your head.  Imagine your client looks at the app and says that they'd rather have it say "Reload" instead of "Retry".  Now you've got to find all the places in your code where you're switching based on "Retry".  What if you have more than one button in your app that say's "Retry", but you only want to change one of them?  Even worse, imagine that you've localized this app, so that the user interface is in a different language depending on the user?  This would never work.

So, you should switch on the buttonIndex here instead.  I don't really like the way alertView works in this respect, because it's not easy to look at the method and understand which button was index 1, but that's what you're stuck with.

In a real project, I like to use this alert view replacement:
https://github.com/sberrevoets/SDCAlertView
Instead of using a delegate call back, when you make your SDCAlertView, you assign a block to each button.  That block gets executed with the button is tapped, so it's much cleaner and clearer all around.

But, with the basic toolkit from Apple, you're stuck with indexing by button number.